src/graph.cpp

```diff
@@ -210,4 +217,52 @@ void Graph::augmentPath(const int &source, const int &target, const double &valu
             }
         }
     }
-}
\ No newline at end of file
+}
+
+//change to return void
+std::vector<Edge*> Graph::deactivateEdges(int numEdges) {
+    int id = 0;
+    int choice = 0;
+    int currentEdges = 0;
+    std::vector<Edge*> deactivatedEdges = {};
+    while(currentEdges < numEdges){
+        id = rand() % vertexSet.size();
+        choice = rand() % vertexSet[id]->getAdj().size();
+        if(vertexSet[id]->getAdj()[choice]->isSelected()){
+           vertexSet[id]->getAdj()[choice]->setSelected(false);
+           deactivatedEdges.push_back(vertexSet[id]->getAdj()[choice]);
+           currentEdges++;
+        }
+    }
+    return deactivatedEdges;
+}
+
+std::vector<Edge*> Graph::deactivateEdges(std::vector<Edge *> Edges) {
+    std::vector<Edge*> deactivatedEdges = {};
+    for(Edge* edge : Edges){
+        edge->setSelected(false);
+        deactivatedEdges.push_back(edge);
```

Comment:
Why create a new deactivatedEdges variable if the Edges you are going to deactivate are exactly the ones already in `Edges`? One way this could work is if you check if the Edge is already deactivated, and make it so deactivatedEdges only returns the edges whose state actually changes. If not, simply returning `Edges` should work.